From d3adc9ff87e6f88b6331485f29045d9c29e3f792 Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Mon, 23 Jun 2014 21:51:52 -0400
Subject: [fs] nfsd: fix FREE_STATEID lockowner leak

Message-id: <1403560314-32525-171-git-send-email-bfields@redhat.com>
Patchwork-id: 84583
O-Subject: [Patch RHEL7 170/172] nfsd4: fix FREE_STATEID lockowner leak
Bugzilla: 1111719
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

27b11428b7de ("nfsd4: remove lockowner when removing lock stateid")
introduced a memory leak.

Cc: stable@vger.kernel.org
Reported-by: Jeff Layton <jeff.layton@primarydata.com>
Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: 48385408b45523d9a432c66292d47ef43efcbb94
Bugzilla: 1111719
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/nfs4state.c b/fs/nfsd/nfs4state.c
index e5197d9..c0d45ce 100644
--- a/fs/nfsd/nfs4state.c
+++ b/fs/nfsd/nfs4state.c
@@ -3743,7 +3743,7 @@ nfsd4_free_lock_stateid(struct nfs4_ol_stateid *stp)
   * correspondance, and we have to delete the lockowner when we
   * delete the lock stateid:
   */
- unhash_lockowner(lo);
+ release_lockowner(lo);
  return nfs_ok;
 }
 
-- 
1.7.1